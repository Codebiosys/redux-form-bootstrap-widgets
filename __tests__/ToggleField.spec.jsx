/* eslint-disable import/first */

import React from 'react';
import { mount } from 'enzyme';
import { omit } from 'lodash';
import { ToggleField } from 'index';

const fieldProps = {
  input: {
    name: 'fieldName',
    onChange: jest.fn(),
    onBlur: jest.fn(),
    onFocus: jest.fn(),
  },
  meta: {
    form: 'formName',
    pristine: false,
  },
  label: 'CheckBox  Test',
  helpText: 'The help Text',
};

const selectedFieldProps = { ...fieldProps };
selectedFieldProps.input = { ...fieldProps.input };
selectedFieldProps.input.value = true;

describe('The Toggle Field', () => {
  let inputWrapper;

  beforeEach(() => {
    inputWrapper = mount(<ToggleField {...fieldProps} />);
  });

  it('renders', () => {
    expect(inputWrapper).toMatchSnapshot();
  });

  it('has a label when there is a label', () => {
    expect(inputWrapper.find('ControlLabel').text()).toEqual(fieldProps.label);
  });

  it('is not checked by default', () => {
    expect(inputWrapper.find(`input[name="${fieldProps.input.name}"]`).prop('checked')).toBe(false);
  });

  it('is checked when the value is true', () => {
    inputWrapper = mount(<ToggleField {...selectedFieldProps} />);
    expect(inputWrapper.find(`input[name="${fieldProps.input.name}"]`).prop('checked')).toBe(true);
  });
  it('triggers the focus when focused', () => {
    inputWrapper.find(`input[name="${fieldProps.input.name}"]`).simulate('focus');
    expect(fieldProps.input.onFocus).toHaveBeenCalledTimes(1);
  });

  it('triggers the blur when blurred', () => {
    inputWrapper.find(`input[name="${fieldProps.input.name}"]`).simulate('blur');
    expect(fieldProps.input.onBlur).toHaveBeenCalledTimes(1);
  });

  it('calls onChange when the toogle is toggled', () => {
    inputWrapper.find(`input[name="${fieldProps.input.name}"]`).simulate('change', { target: { checked: true } });
    expect(fieldProps.input.onChange).toHaveBeenCalledTimes(1);
    expect(fieldProps.input.onChange).toHaveBeenCalledWith(true);
  });

  it('calls onChange when the toogle is toggled off', () => {
    inputWrapper.find(`input[name="${fieldProps.input.name}"]`).simulate('change', { target: { checked: false } });
    expect(fieldProps.input.onChange).toHaveBeenCalledTimes(1);
    expect(fieldProps.input.onChange).toHaveBeenCalledWith(false);
  });

  it('calls custom validator when the toogle is toggled', () => {
    const customValidator = jest.fn(() => ({ validationState: null, errorMessage: null }));
    const validatorProps = { ...fieldProps, customValidation: customValidator };
    const inputWrapperValidated = mount(<ToggleField {...validatorProps} />);

    inputWrapperValidated.find(`input[name="${fieldProps.input.name}"]`).simulate('change', { target: { checked: true } });
    expect(customValidator).toHaveBeenCalledTimes(1);
  });
});
