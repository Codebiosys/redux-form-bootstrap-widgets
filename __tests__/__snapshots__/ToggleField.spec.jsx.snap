// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Toggle Field renders 1`] = `
<ToggleField
  helpText="The help Text"
  input={
    Object {
      "name": "fieldName",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
    }
  }
  label="CheckBox  Test"
  meta={
    Object {
      "form": "formName",
      "pristine": false,
    }
  }
>
  <FormGroup
    bsClass="form-group"
    controlId="fieldName"
  >
    <div
      className="form-group"
    >
      <Component
        label="CheckBox  Test"
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          <label
            className="control-label"
            htmlFor="fieldName"
          >
            CheckBox  Test
          </label>
        </ControlLabel>
      </Component>
      <InputGroup
        bsClass="input-group"
      >
        <span
          className="input-group"
        >
          <Toggle
            checked={false}
            icons={
              Object {
                "checked": <Unknown />,
                "unchecked": <Unknown />,
              }
            }
            name="fieldName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
          >
            <div
              className="react-toggle"
              onClick={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="react-toggle-track"
              >
                <div
                  className="react-toggle-track-check"
                >
                  <Component>
                    <svg
                      height="11"
                      viewBox="0 0 14 11"
                      width="14"
                    >
                      <title>
                        switch-check
                      </title>
                      <path
                        d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                        fill="#fff"
                        fillRule="evenodd"
                      />
                    </svg>
                  </Component>
                </div>
                <div
                  className="react-toggle-track-x"
                >
                  <Component>
                    <svg
                      height="10"
                      viewBox="0 0 10 10"
                      width="10"
                    >
                      <title>
                        switch-x
                      </title>
                      <path
                        d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                        fill="#fff"
                        fillRule="evenodd"
                      />
                    </svg>
                  </Component>
                </div>
              </div>
              <div
                className="react-toggle-thumb"
              />
              <input
                checked={false}
                className="react-toggle-screenreader-only"
                name="fieldName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
              />
            </div>
          </Toggle>
        </span>
      </InputGroup>
      <HelpBlock
        bsClass="help-block"
      >
        <span
          className="help-block"
        >
          The help Text
        </span>
      </HelpBlock>
    </div>
  </FormGroup>
</ToggleField>
`;
